pkg/printers/template_flags.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// TemplatePrintFlags provides default flags necessary for template printing.
+// Given the following flag values, a printer can be requested that knows
+// how to handle printing based on these values.
+type TemplatePrintFlags struct {
+	// indicates if it is OK to ignore missing keys for rendering
+	// an output template.
+	AllowMissingKeys *bool
+	TemplateArgument *string
+}
+
+func (f *TemplatePrintFlags) Validate() error {
+	if f.TemplateArgument == nil || len(*f.TemplateArgument) == 0 {
+		return fmt.Errorf("missing --template value")
+	}
+	return nil
+}
+
+// ToPrinter receives an outputFormat and returns a printer capable of
+// handling --template format printing.
+// Returns false if the specified outputFormat does not match a template format.
+func (f *TemplatePrintFlags) ToPrinter(outputFormat string) (ResourcePrinter, bool, error) {
+	if f.TemplateArgument == nil {
+		return nil, true, fmt.Errorf("missing --template argument")
+	}
+	templateArg := *f.TemplateArgument
+
+	// if no output format is explicitly given via --output
+	// but a template value is provided via --template=...
+	// default --output format to go-template.
+	if len(outputFormat) == 0 && len(templateArg) > 0 {
+		outputFormat = "go-template"
+	}
+
+	// templates are logically optional for specifying a format.
+	// this allows a user to specify a template format value
+	// as --output=go-template=
+	templateFormats := map[string]bool{
```

Comment:
This should defined outside this method. It's a constant.

Reply:
Thanks, done